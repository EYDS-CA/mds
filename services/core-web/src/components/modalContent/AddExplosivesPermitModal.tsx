import React, { FC } from "react";
import { IOption, IGroupedDropdownList, IExplosivesPermitDocument } from "@mds/common/interfaces";
import ExplosivesPermitFormNew from "@/components/Forms/ExplosivesPermit/ExplosivesPermitFormNew";

interface ExplosivesPermitModalProps {
  title: string;
  initialValues: any;
  documents: IExplosivesPermitDocument[];
  mineGuid: string;
  isApproved: boolean;
  documentTypeDropdownOptions: IOption[];
  isPermitTab: boolean;
  inspectors: IGroupedDropdownList[];
  isProcessed: boolean;
  isAmendment: boolean;
  onSubmit: (values) => void | Promise<void>;
}

export const AddExplosivesPermitModal: FC<ExplosivesPermitModalProps> = (props) => (
  <div>
    <ExplosivesPermitFormNew {...props} />
  </div>
);

export default AddExplosivesPermitModal;
