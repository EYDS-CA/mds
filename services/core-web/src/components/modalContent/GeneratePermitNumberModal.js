import React from "react";
import PropTypes from "prop-types";
import { Alert } from "antd";
import GeneratePermitNumberForm from "@/components/Forms/permits/GeneratePermitNumberForm";

const propTypes = {
  onSubmit: PropTypes.func.isRequired,
  signature: PropTypes.string.isRequired,
  title: PropTypes.string,
};

const defaultProps = {
  title: "Generate Permit Number",
};

export const GeneratePermitNumberModal = (props) => {
  return (
    <div>
      {!props.signature && (
        <>
          <Alert
            message="Signature needed to generate."
            description="The signature for the Issuing Inspector has not been provided."
            type="error"
            showIcon
          />
          <br />
        </>
      )}
      <GeneratePermitNumberForm
        onSubmit={props.onSubmit}
        disabled={!props.signature}
      />
    </div>
  );
};

GeneratePermitNumberModal.propTypes = propTypes;
GeneratePermitNumberModal.defaultProps = defaultProps;

export default GeneratePermitNumberModal;
