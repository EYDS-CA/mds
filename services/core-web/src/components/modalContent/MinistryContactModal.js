import React from "react";
import PropTypes from "prop-types";
import MinistryContactForm from "@/components/Forms/MinistryContacts/MinistryContactForm";

const propTypes = {
  closeModal: PropTypes.func.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
  title: PropTypes.string.isRequired,
};

const defaultProps = {};

export const MinistryContactModal = (props) => {
  return (
    <div>
      <MinistryContactForm {...props} />
    </div>
  );
};

MinistryContactModal.propTypes = propTypes;
MinistryContactModal.defaultProps = defaultProps;

export default MinistryContactModal;
