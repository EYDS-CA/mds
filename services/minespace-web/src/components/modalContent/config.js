import AddVarianceModal from "@/components/modalContent/variances/AddVarianceModal";
import ViewVarianceModal from "@/components/modalContent/variances/ViewVarianceModal";
import EditVarianceModal from "@/components/modalContent/variances/EditVarianceModal";
import AddTailingsModal from "@/components/modalContent/tailing/AddTailingsModal";
import AddContactModal from "@/components/modalContent/contacts/AddContactModal";
import AddNoticeOfDepartureModal from "@/components/modalContent/noticeOfDeparture/AddNoticeOfDepartureModal";
import ViewNoticeOfDepartureModal from "@/components/modalContent/noticeOfDeparture/ViewNoticeOfDepartureModal";
import EditNoticeOfDepartureModal from "@/components/modalContent/noticeOfDeparture/EditNoticeOfDepartureModal";
import ImportIRTSuccessModal from "@/components/modalContent/informationRequirementsTable/ImportIRTSuccessModal";
import ImportIRTErrorModal from "@/components/modalContent/informationRequirementsTable/ImportIRTErrorModal";
import ViewFileHistoryModal from "./informationRequirementsTable/ViewFileHistoryModal";
import UploadIncidentDocumentModal from "@/components/modalContent/incidents/UploadIncidentDocumentModal";
import CreateInvitationModal from "@/components/modalContent/verifiableCredentials/CreateInvitationModal";

export const modalConfig = {
  ADD_VARIANCE: AddVarianceModal,
  VIEW_VARIANCE: ViewVarianceModal,
  EDIT_VARIANCE: EditVarianceModal,
  ADD_TAILINGS: AddTailingsModal,
  ADD_CONTACT: AddContactModal,
  ADD_NOTICE_OF_DEPARTURE: AddNoticeOfDepartureModal,
  VIEW_NOTICE_OF_DEPARTURE: ViewNoticeOfDepartureModal,
  EDIT_NOTICE_OF_DEPARTURE: EditNoticeOfDepartureModal,
  IMPORT_IRT_SUCCESS: ImportIRTSuccessModal,
  IMPORT_IRT_FAILURE: ImportIRTErrorModal,
  VIEW_FILE_HISTORY: ViewFileHistoryModal,
  UPLOAD_INCIDENT_DOCUMENT: UploadIncidentDocumentModal,
  VC_WALLET_INVITATION: CreateInvitationModal,
};

export default modalConfig;
