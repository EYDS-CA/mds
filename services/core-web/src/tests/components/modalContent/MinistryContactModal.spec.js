import React from "react";
import { shallow } from "enzyme";
import { MinistryContactModal } from "@/components/modalContent/MinistryContactModal";

const dispatchProps = {};
const props = {};

const setupDispatchProps = () => {
  dispatchProps.oSubmit = jest.fn();
  dispatchProps.closeModal = jest.fn();
};

const setupProps = () => {
  props.initialValues = {};
  props.regionDropdownOptions = [];
  props.MinistryContactTypes = [];
  props.isEdit = true;
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("MinistryContactModal", () => {
  it("renders properly", () => {
    const component = shallow(<MinistryContactModal {...dispatchProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
