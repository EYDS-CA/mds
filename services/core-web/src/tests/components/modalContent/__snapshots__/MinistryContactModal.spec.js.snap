// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinistryContactModal renders properly 1`] = `
<div>
  <Connect(ReduxForm)
    MinistryContactTypes={Array []}
    closeModal={[MockFunction]}
    initialValues={Object {}}
    isEdit={true}
    oSubmit={[MockFunction]}
    regionDropdownOptions={Array []}
  />
</div>
`;
